pkg/envtest/binaries_test.go

```diff
@@ -28,13 +28,93 @@ import (
 	"os"
 	"path"
 	"runtime"
+	"strings"
+	"testing"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"github.com/onsi/gomega/ghttp"
 	"sigs.k8s.io/yaml"
 )
 
+func TestInterpretKubernetesVersion(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name  string
+		in    []string
+		exact bool
+		major uint
+		minor uint
+	}{
+		{
+			name:  `SemVer and "v" prefix are exact`,
+			exact: true,
+			in: []string{
+				"1.2.3", "v1.2.3", "v1.30.2", "v1.31.0-beta.0", "v1.33.0-alpha.2",
+			},
+		},
+		{
+			name: "leading zeroes are not a version",
+			in: []string{
+				"01.2.0", "00001.2.3", "1.2.03", "v01.02.0003",
+			},
+		},
+		{
+			name: "weird stuff is not a version",
+			in: []string{
+				"asdf", "version", "vegeta4", "the.1", "2ne1", "=7.8.9", "10.x", "*",
+				"0.0001", "1.00002", "v1.2anything", "1.2.x", "1.2.z", "1.2.*",
+			},
+		},
+		{
+			name: "one number is not a version",
+			in: []string{
+				"1", "v1", "v001", "1.", "v1.", "1.x",
+			},
+		},
+		{
+			name:  "two numbers are a release series",
+			major: 0, minor: 1,
+			in: []string{"0.1", "v0.1"},
```

Comment:
These testcases would IMHO be less confusing to read if we ordered the properties 1. name, 2. input 3. expected outputs , ther than 1.name, 2. expected outputs, 3.input